edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/DiagnosticConfig.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using Microsoft.Extensions.Configuration;
+
+    public class DiagnosticConfig
+    {
+        public bool Enabled;
+        public string MetricsStoragePath;
+        public TimeSpan ScrapeInterval;
+        public TimeSpan UploadInterval;
+
+        public DiagnosticConfig(bool enabled, string storagePath, IConfiguration configuration)
+        {
+            this.Enabled = enabled;
```

Comment:
Parameter validation, e.g. do we check if storage path exists, or configuration is not null.

Reply:
storage path will be created if it doesn't exist. I could check configuration for not null, but it will throw in the same place since I immediately use it. I don;t see a point in checking.

Reply:
it is better to do validation proactive with proper exception message.  For this case, you should throw argumentnullexception rather than nullreferenceexception; when throw argumentnullexception you should provide which argument causes this issue for easy troubleshooting.